myapp/app/controllers/tasks_controller.rb

```diff
@@ -3,8 +3,14 @@
 class TasksController < ApplicationController
   before_action :set_task, only: %i[show edit update destroy]
 
+  VALID_SORT_COLUMNS = %w[created_at deadline].freeze
+  VALID_SORT_DIRECTIONS = %w[asc desc].freeze
+
+  # GET /tasks
   def index
-    @tasks = Task.all.order(created_at: :desc)
+    @sort_by = VALID_SORT_COLUMNS.include?(params[:sort_by]) ? params[:sort_by] : 'created_at'
+    @sort_direction = VALID_SORT_DIRECTIONS.include?(params[:sort_direction]) ? params[:sort_direction] : 'asc'
```

Comment:
Do @sort_by and @sort_direction need to be instance variables?

If not particularly necessary, let's make them local variables (change @sort_by to sort_by). We can make them instance variables when needed.

Other engineers who see instance variables might assume they are used in view files or elsewhere, which could lead to misunderstandings while reading the source code.

Reply:
Thank you for your comment, but I changed it to a local variable and it won't work.

Reply:
Thank you I changed it